cmd/dump/cli-reference.go

```diff
@@ -0,0 +1,186 @@
+
+			linkHandler := func(name string) string {
+				if name == "furyctl.md" {
+					return "../index.md"
+				}
+
+				for _, command := range cmd.Root().Commands() {
+					basename := strings.ReplaceAll(command.CommandPath(), " ", "_") + ".md"
+					if basename == name {
+						return fmt.Sprintf("../%s/index.md", command.Name())
+					}
+				}
+
+				return name
+			}
+
+			cmd.Root().DisableAutoGenTag = true
+
+			if err := GenMarkdownCustom(cmd.Root(), mainFile, linkHandlerRoot); err != nil {
+				return fmt.Errorf("failed to generate CLI reference: %w", err)
+			}
+			for _, command := range cmd.Root().Commands() {
+				outputPath := fmt.Sprintf("%s/%s", outputFolder, command.Name())
+				if err := os.MkdirAll(outputPath, outputFolderPerms); err != nil {
+					return fmt.Errorf("failed to create output folder: %w", err)
+				}
+				err := GenMarkdownTreeCustom(command, outputPath, dummyFilePrepender, linkHandler)
+				if err != nil {
+					return fmt.Errorf("failed to generate CLI reference for command %s: %w", command.Name(), err)
+				}
+				if err := os.Rename(fmt.Sprintf("%s/furyctl_%s.md", outputPath, command.Name()), filepath.Join(outputPath, "index.md")); err != nil {
+					return fmt.Errorf("failed to rename CLI reference file for command %s: %w", command.Name(), err)
+				}
+				// We need to this, because the markdown file generated by cobra has a code block that is not wrapped in triple backticks.
+				if command.Name() == "completion" {
+					if err := escapeCodeBlock(filepath.Join(outputPath, "index.md")); err != nil {
+						return fmt.Errorf("failed to escape code block in CLI reference for command %s: %w", command.Name(), err)
+					}
+				}
+			}
+
+			logrus.Infof("Markdown CLI reference successfully exported to %s", outputFolder)
+
+			cmdEvent.AddSuccessMessage("CIL Reference generated successfully")
+			tracker.Track(cmdEvent)
+
+			return nil
+		},
+	}
+
+	dumpCLIReferenceCmd.Flags().Bool("no-overwrite", true, "Do not overwrite existing files. Will exit if the output folder already exists")
+	dumpCLIReferenceCmd.Flags().StringP("workdir", "w", "", "Working directory to use for the output folder. Default is the current working directory")
+
+	return dumpCLIReferenceCmd
+}
+
+func getDumpCliReferenceCmdFlags() CliReferenceCmdFlags {
+	return CliReferenceCmdFlags{
+		NoOverwrite: viper.GetBool("no-overwrite"),
+		Workdir:     viper.GetString("workdir"),
+	}
+}
+
+func escapeCodeBlock(path string) error {
+	// Escape code blocks in the file surrounding them with triple backticks.
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %w", path, err)
+	}
+
+	reUnix := regexp.MustCompile(`(.+)\$ (.+)`)
+	rePS := regexp.MustCompile(`(.+)PS> (.+)`)
+	escapedContent := string(content)
+	escapedContent = reUnix.ReplaceAllString(escapedContent, "${1}```shell\n${1}$$ ${2}\n${1}```")
+	escapedContent = rePS.ReplaceAllString(escapedContent, "${1}```powershell\n${1}PS> ${2}\n${1}```")
```

Comment:
this is a fix for a linting error inside the generated MD where codeblocks are not properly sorrounded by blank lines (e.g.: `furyctl completion`), but it makes sense to maintain that styling for the CLI help messages:

```suggestion
	escapedContent = rePS.ReplaceAllString(escapedContent, "${1}```powershell\n${1}PS> ${2}\n${1}```")
        reFencedNewLineFix := regexp.MustCompile(":\\n(.+)```")
        escapedContent = reFencedNewLineFix.ReplaceAllString(escapedContent, ":\n\n${1}```")
```